TimeOfFlight/src/TOFEstimators.cc

```diff
@@ -33,442 +43,595 @@
 #include <gsl/gsl_rng.h>
 #include <gsl/gsl_randist.h>
 
-//---- ROOT -----
-#include "TH2F.h"
-
 using namespace lcio ;
 using namespace marlin ;
 using namespace TOFUtils ;
 
-TOFEstimators aTOFEstimators ;
+// Speed of light in mm/ns = 299.79246
+#define SPEED_OF_LIGHT TMath::C()/1.e6
```

Comment:
Not better than before. You make a division everytime you use this constant. Use `CLHEP::c_light` instead with the `CLHEP/Units/PhysicalConstants.h` include please

Reply:
Compiler can probably figure it out in this case. ;)

Nevertheless, using CLHEP you should be able to do
```suggestion
static constexpr double SPEED_OF_LIGHT = CLHEP::c_light;
```

or replace the `SPEED_OF_LIGHT` constant completely by `CLHEP::c_light`. **Beware that c_light` already is in mm/ns as per CLHEP convention**, but if you want to be really sure that this stays that way, even if CLHEP decides to change its basis of units do
```suggestion
static constexpr double SPEED_OF_LIGHT = CLHEP::c_light * CLHEP::mm / CLHEP::ns;
```
and you have explicitly stated what your speed of light is, even with units.

Reply:
Is it in mm/ns? if I look in the [header](https://www-zeuthen.desy.de/geant4/clhep-2.0.4.3/PhysicalConstants_8h-source.html) file its in m/s. Am I wrong?

```
 //
00061 // c   = 299.792458 mm/ns
00062 // c^2 = 898.7404 (mm/ns)^2 
00063 //
00064 static const double c_light   = 2.99792458e+8 * m/s;
00065 static const double c_squared = c_light * c_light;
00066 
```

Reply:
First the header you are looking at is an older version. The version that comes with v02-02 is 2.4.4.3

There the definition is 
https://gitlab.cern.ch/CLHEP/CLHEP/-/blob/CLHEP_2_3_4_3/Units/Units/PhysicalConstants.h#L56

So, still m / s. But if you then check how m and s are defined you will see, that the base units of CLHEP are indeed mm and ns, and the units are defined accordingly:
https://gitlab.cern.ch/CLHEP/CLHEP/-/blob/CLHEP_2_3_4_3/Units/Units/SystemOfUnits.h#L72
https://gitlab.cern.ch/CLHEP/CLHEP/-/blob/CLHEP_2_3_4_3/Units/Units/SystemOfUnits.h#L140

So, if you need a value in specific units, you can transform the numerical value into the CLHEP system by using their units as well.